### PR TITLE
owncloudclient: update to 5.2.1.

### PR DESCRIPTION
**Pull request by tornaria**
#### Testing the changes
- I tested the changes in this PR: **briefly**

I've been using 5.0.0 for a while, and now updated to 5.1.0.

Note that kde integration (#45352) has been removed.
There is an external project: https://github.com/owncloud/client-desktop-shell-integration-dolphin but I don't use kde and I wouldn't be able to test at all.

It was necessary to build `qtkeychain-qt6` which I copied with obvious changes from `qtkeychain-qt5`. Should this new pkg keep @yopito as maintainer?

### CODE REVIEW

**Comment by tornaria** [comment]
I wonder if `-DKDE_INSTALL_QTPLUGINDIR=lib/qt6/plugins` and `-DECM_MKSPECS_INSTALL_DIR=/usr/lib/qt6/mkspecs` should be in `common/build-style/cmake.sh` just as `-DQT_HOST_PATH=/usr` and `-DQT_HOST_PATH_CMAKE_DIR=/usr/lib/cmake` are.

**Comment by tornaria** [comment]
Here:
```
--- a/src/sage/features/threejs.py
+++ b/src/sage/features/threejs.py
@@ -61,4 +61,4 @@ class Threejs(StaticFile):
 
 
 def all_features():
-    return [Threejs()]
+    return []
```

File: srcpkgs/libre-graph-api-cpp-qt-client/template [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # Template file for 'libre-graph-api-cpp-qt-client'
 pkgname=libre-graph-api-cpp-qt-client
 version=1.0.4
-revision=1
+revision=2
 build_wrksrc=client
 build_style=cmake
-hostmakedepends="qt5-qmake qt5-host-tools"
-makedepends="qt5-devel"
+hostmakedepends="qt6-tools-devel"
```
Comment (ahesford) [inline srcpkgs/libre-graph-api-cpp-qt-client/template:7]: Please reconcile with b0a4ed8d2158

File: srcpkgs/owncloudclient/template [hunk 2]
```diff
@@ -1,33 +1,30 @@
...
-configure_args="-Wno-dev -DWITH_AUTO_UPDATER=OFF"
-hostmakedepends="pkg-config extra-cmake-modules qt5-host-tools qt5-tools-devel
- kcoreaddons"
-makedepends="qtkeychain-qt5-devel sqlite-devel qt5-declarative-devel kio-devel
- qt5-tools-devel qt5-plugin-odbc qt5-plugin-tds qt5-plugin-pgsql qt5-plugin-mysql
- qt5-plugin-sqlite libre-graph-api-cpp-qt-client libcloudproviders-devel"
-depends="qt5-plugin-sqlite qt5-svg"
+configure_args="-Wno-dev -DWITH_AUTO_UPDATER=OFF
+ -DKDE_INSTALL_QTPLUGINDIR=lib/qt6/plugins"
+hostmakedepends="pkg-config extra-cmake-modules qt6-tools-devel"
```
Comment (ahesford) [inline srcpkgs/owncloudclient/template:8]: Please reconcile with b0a4ed8d2158

**Review by ahesford — CHANGES_REQUESTED** [review]

**Comment by tornaria** [comment]
@ahesford it makes a lot of sense, sorry I missed that. I did it in the four templates. I tested locally the cross-build on aarch64 to make sure I got the `hostmakedepends` right (sometimes `qt6-base` suffices, other times `qt6-tools` is also necessary).

**Pull request merged**